obs-packaging/Dockerfile

```diff
@@ -1,8 +1,10 @@
-ARG SUSE_VERSION=${SUSE_VERSION:-42.3}
+ARG SUSE_VERSION=${SUSE_VERSION:-15.1}
 FROM opensuse/leap:${SUSE_VERSION}
 
 
 # Get OBS client, plugins and dependencies
+#curl-7.59.0-lp150.1.1.<arch> requires libcurl4 = 7.59.0. Default libcurl4-7.60.0 present.
+RUN zypper -v -n install --oldpackage libcurl4=7.59.0-lp150.1.1
```

Comment:
I don't understand this @nitkon , while libcurl4 needs to be manually installed, and why you need for force a version?

Reply:
Hi @marcov ,  `libcurl4-7.60.0` is already present in the base image. When installing curl, it looks for `libcurl4 = 7.59.0.`, which if not present, fails.
```
Resolving package dependencies...
Force resolution: No

Problem: curl-7.59.0-lp150.1.1.ppc64le requires libcurl4 = 7.59.0, but this requirement cannot be provided
  not installable providers: libcurl4-7.59.0-lp150.1.1.ppc64le[repo-oss]
 Solution 1: downgrade of libcurl4-7.60.0-lp150.2.18.1.ppc64le to libcurl4-7.59.0-lp150.1.1.ppc64le
 Solution 2: do not install curl-7.59.0-lp150.1.1.ppc64le
 Solution 3: break curl-7.59.0-lp150.1.1.ppc64le by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/3/c] (c): c
The command '/bin/sh -c zypper -v -n install osc-plugin-install vim curl bsdtar git sudo' returned a non-zero code: 4
```

Reply:
Does that happen even if run this?
```
$ docker run -it --rm opensuse/leap:15.1 zypper in -y curl
```

Reply:
Also, could you start a clean `opensuse/leap:15.1` container and paste the output of:
```
$ rpm -q libcurl4
$ zypper info --requires curl
```

Reply:
Hi @marcov :
```
root@soe13:~# docker run -t --rm opensuse/leap:15.1 zypper in -y curl
Retrieving repository 'openSUSE-Leap-15.0-Oss' metadata ................................................................................................[done]
Building repository 'openSUSE-Leap-15.0-Oss' cache .....................................................................................................[done]
Retrieving repository 'openSUSE-Leap-15.0-Update' metadata .............................................................................................[done]
Building repository 'openSUSE-Leap-15.0-Update' cache ..................................................................................................[done]
Loading repository data...
Reading installed packages...
Resolving package dependencies...

Problem: curl-7.59.0-lp150.1.1.ppc64le requires libcurl4 = 7.59.0, but this requirement cannot be provided
  not installable providers: libcurl4-7.59.0-lp150.1.1.ppc64le[repo-oss]
 Solution 1: downgrade of libcurl4-7.60.0-lp150.2.18.1.ppc64le to libcurl4-7.59.0-lp150.1.1.ppc64le
 Solution 2: do not install curl-7.59.0-lp150.1.1.ppc64le
 Solution 3: break curl-7.59.0-lp150.1.1.ppc64le by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/3/c] (c): c
root@soe13:~#
```

```
root@soe13:~# docker run -it opensuse/leap:15.1 /bin/bash
4f45cce3be0e:/ # rpm -q libcurl4
libcurl4-7.60.0-lp150.2.18.1.ppc64le
4f45cce3be0e:/ # zypper info --requires curl
Retrieving repository 'openSUSE-Leap-15.0-Oss' metadata ---------------------------------------------------------------------------------------------------[-]
Retrieving repository 'openSUSE-Leap-15.0-Oss' metadata ................................................................................................[done]
Building repository 'openSUSE-Leap-15.0-Oss' cache .....................................................................................................[done]
Retrieving repository 'openSUSE-Leap-15.0-Update' metadata ------------------------------------------------------------------------------------------------[-]
Retrieving repository 'openSUSE-Leap-15.0-Update' metadata .............................................................................................[done]
Building repository 'openSUSE-Leap-15.0-Update' cache ..................................................................................................[done]
Loading repository data...
Reading installed packages...


Information for package curl:
-----------------------------
Repository     : openSUSE-Leap-15.0-Oss                
Name           : curl                                  
Version        : 7.59.0-lp150.1.1                      
Arch           : ppc64le                               
Vendor         : openSUSE                              
Installed Size : 583.2 KiB                             
Installed      : No                                    
Status         : not installed                         
Source package : curl-7.59.0-lp150.1.1.src             
Summary        : A Tool for Transferring Data from URLs
Description    :                                       
    Curl is a client to get documents and files from or send documents to a
    server using any of the supported protocols (HTTP, HTTPS, FTP, FTPS,
    TFTP, DICT, TELNET, LDAP, or FILE). The command is designed to work
    without user interaction or any kind of interactivity.
Requires       : [9]                                   
    libc.so.6(GLIBC_2.17)(64bit)
    libpthread.so.0()(64bit)
    libpthread.so.0(GLIBC_2.17)(64bit)
    libz.so.1()(64bit)
    libcrypto.so.1.1()(64bit)
    libcrypto.so.1.1(OPENSSL_1_1_0)(64bit)
    libcurl.so.4()(64bit)
    libmetalink.so.3()(64bit)
    libcurl4 = 7.59.0
```

Reply:
Dumb question: have you tried to run a `zypper ref` before doing the install? I just checked in the ppc repo and it is expected to have up to date curl and libcurl4:

https://download.opensuse.org/ports/ppc/distribution/leap/15.1/repo/oss/ppc64le/

Reply:
@marcov : Result is the same:
```
Sending build context to Docker daemon  2.048kB
Step 1/5 : ARG SUSE_VERSION=${SUSE_VERSION:-15.1}
Step 2/5 : FROM opensuse/leap:${SUSE_VERSION}
 ---> 1e6d631304a8
Step 3/5 : RUN zypper ref
 ---> Running in f61e05b7b4df
Retrieving repository 'openSUSE-Leap-15.0-Oss' metadata [...........done]
Building repository 'openSUSE-Leap-15.0-Oss' cache [....done]
Retrieving repository 'openSUSE-Leap-15.0-Update' metadata [..............done]
Building repository 'openSUSE-Leap-15.0-Update' cache [....done]
All repositories have been refreshed.
Removing intermediate container f61e05b7b4df
 ---> 577a7cbfb1b9
Step 4/5 : RUN zypper -v -n install osc-plugin-install vim curl bsdtar git sudo
 ---> Running in 055895a80a58
Verbosity: 1
Entering non-interactive mode.
Non-option program arguments: 'osc-plugin-install' 'vim' 'curl' 'bsdtar' 'git' 'sudo' 
Initializing Target
Checking whether to refresh metadata for openSUSE-Leap-15.0-Oss
Checking whether to refresh metadata for openSUSE-Leap-15.0-Update
Loading repository data...
Reading installed packages...
Force resolution: No
Selecting 'vim-8.0.1568-lp150.3.1.ppc64le' from repository 'openSUSE-Leap-15.0-Oss' for installation.
Selecting 'curl-7.59.0-lp150.1.1.ppc64le' from repository 'openSUSE-Leap-15.0-Oss' for installation.
Selecting 'git-2.16.3-lp150.1.2.ppc64le' from repository 'openSUSE-Leap-15.0-Oss' for installation.
Selecting 'sudo-1.8.22-lp150.2.3.ppc64le' from repository 'openSUSE-Leap-15.0-Oss' for installation.
Selecting 'bsdtar-3.3.2-lp150.1.3.ppc64le' from repository 'openSUSE-Leap-15.0-Oss' for installation.
Selecting 'osc-plugin-install-0.25-lp150.2.3.noarch' from repository 'openSUSE-Leap-15.0-Oss' for installation.
Resolving package dependencies...
Force resolution: No

Problem: curl-7.59.0-lp150.1.1.ppc64le requires libcurl4 = 7.59.0, but this requirement cannot be provided
  not installable providers: libcurl4-7.59.0-lp150.1.1.ppc64le[repo-oss]
 Solution 1: downgrade of libcurl4-7.60.0-lp150.2.18.1.ppc64le to libcurl4-7.59.0-lp150.1.1.ppc64le
 Solution 2: do not install curl-7.59.0-lp150.1.1.ppc64le
 Solution 3: break curl-7.59.0-lp150.1.1.ppc64le by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/3/c] (c): c
The command '/bin/sh -c zypper -v -n install osc-plugin-install vim curl bsdtar git sudo' returned a non-zero code: 4
```

Reply:
Wait: why do have 15.0 repos set up in a 15.1 container image?
```
Retrieving repository 'openSUSE-Leap-15.0-Oss' metadata [...........done]
Building repository 'openSUSE-Leap-15.0-Oss' cache [....done]
Retrieving repository 'openSUSE-Leap-15.0-Update' metadata [..............done]
Building repository 'openSUSE-Leap-15.0-Update' cache [....done]
```

Reply:
@nitkon, I'll try to get that fixed internally. Can we hold this  PR just a bit?

Reply:
Yea that's weird. It would be great if we can get this fixed. Will hold the PR till then. You can put a DNM label on this

Reply:
Let's see if the image is fixed quickly.
If not, @nitkon I would prefer as solution to change the oss repo URL instead of pinning the libcurl version, using as base URL: `http://download.opensuse.org/distribution/leap/$releasever/repo/oss/`

Thank you!

Reply:
@nitkon, the opensuse/leap:15.1 image is now using the correct repo URL.
Can you please remove this line and try?

Reply:
@marcov: Thanks. Repushed.